src/app/view/components/dot-navigation/dot-navigation.component.ts

```diff
@@ -12,13 +12,30 @@ import { DotNavigationService } from './dot-navigation.service';
 })
 export class DotNavigationComponent implements OnInit {
     menu: Observable<DotMenu[]>;
+    @Output() onMenuLinkClicked = new EventEmitter<Event>();
 
-    constructor(private dotNavigationService: DotNavigationService) {}
+    constructor(private dotNavigationService: DotNavigationService, private el: ElementRef) {}
 
     ngOnInit() {
         this.menu = this.dotNavigationService.items$;
     }
 
+    /**
+     * Respond to document events and collapse the sidenav if is clicked outside
```

Comment:
Check this service ```IframeOverlayService``` and the way we use it... I think this could be a good fit for this.